PetaPoco/Database.cs

```diff
@@ -624,6 +624,10 @@ private void AddParam(IDbCommand cmd, object value, PropertyInfo pi)
 
         private void SetParameterProperties(IDbDataParameter p, object value, PropertyInfo pi)
         {
+            var columnAttribInfo = pi?.GetCustomAttributes(typeof(ColumnAttribute), true).FirstOrDefault() as ColumnAttribute;
```

Comment:
This seems a little kludgy to me. My hunch is to say that what's really wanted is to get the Ansi and DT2 properties onto `PocoColumn`, and then change calls to `AddParam` and `SetParameterProperties` so that they pass in the `PocoColumn` instead of `PropertyInfo`. (There would still be access to the `PropertyInfo` via `PocoColumn.PropertyInfo`, but now we can also check Ansi and DT2 directly.) 

@pleb What do you think?

Reply:
I need this changes because my "old" DataBase Model has AnsiString/VARCHAR as PK's

Reply:
I'm not sure what you mean by that. My point is that I don't like the idea of digging into `PropertyInfo` here to get the attributes. If this is going to be handled by attributes, then I think it might be better to have the attributes read as part of `PocoColumn` and then pass that in. Take a look at the way `ForceToUtc` is handled; I think something similar should work.

Reply:
added PocoColumn as param

Reply:
I see that you've added `PocoColumn` as a parameter, but I think you can now remove the `PropertyInfo` parameter -- it can be accessed as `pc.PropertyInfo` and so doesn't need to be passed in explicitly.

Reply:
changed in new commt